packages/atomic/src/components/atomic-breadcrumb-manager/atomic-breadcrumb-manager.tsx

```diff
@@ -112,82 +103,85 @@ export class AtomicBreadcrumbManager implements InitializableComponent {
         >
           {label}:
         </span>
-        {children}
+        <ul part="breadcrumbs" class="flex flex-wrap">
+          {children}
+        </ul>
       </li>
     );
   }
 
-  private getBreadcrumbValueWrapper(children: VNode[]) {
+  private getBreadcrumbValueContainer(children: VNode[]) {
     return <li class="mr-3">{children}</li>;
   }
 
-  private getBreadcrumbValues(
-    breadcrumb: Breadcrumb<BaseFacetValue & {value: string}>
-  ) {
+  private getBreadcrumbValues(breadcrumb: BreadcrumbFacet) {
     const {breadcrumbsToShow, moreButton} = this.collapsedBreadcrumbsHandler(
       breadcrumb
     );
     const renderedBreadcrumbs = breadcrumbsToShow.map((breadcrumbValue) =>
-      this.getBreadcrumbValueWrapper(
+      this.getBreadcrumbValueContainer(
         this.getBreadcrumbValue(breadcrumbValue.value.value, breadcrumbValue)
       )
     );
 
-    return this.getBreadcrumbWrapper(
-      breadcrumb.facetId,
-      breadcrumb.field,
-      <ul part="breadcrumbs" class="flex flex-wrap">
-        {[...renderedBreadcrumbs, moreButton]}
-      </ul>
-    );
+    return [...renderedBreadcrumbs, moreButton];
   }
 
   private get facetBreadcrumbs() {
     return this.breadcrumbManagerState.facetBreadcrumbs.map((breadcrumb) => {
-      return this.getBreadcrumbValues(breadcrumb);
+      const {label} = this.bindings.store.get('facets')[breadcrumb.facetId];
```

Comment:
This could, in theory, be null, right ?

Also the store accessor *seems* weakly typed (the `.get('facets')`).

Could we enforce some type of enum to access the store values ? Ensuring better type safety ?

Reply:
> This could, in theory, be null, right ?

I've thought about it and I'm not completely sure.
Facets are registered in headless by the atomic facet components which at the same time set the label to the Atomic store.
The breadcrumbs access the facets in headless to return the value. If it's null it would be a bug on our part 🤔